dynawo/sources/Models/CPP/Controls/Voltage/ModelVoltageMeasurementsUtilities/DYNModelVoltageMeasurementsUtilities.cpp

```diff
@@ -0,0 +1,348 @@
+// for power systems.
+//
+
+/**
+ * @file  DYNModelVoltageMeasurementsUtilities.cpp
+ *
+ * @brief Measurements utility model
+ *
+ * Model to compute some aggregations of the voltages connected to the (sub)network.
+ * Only running models are taken into account for the computations.
+ *
+ */
+
+#include <sstream>
+#include <vector>
+#include "PARParametersSet.h"
+
+#include "DYNModelVoltageMeasurementsUtilities.h"
+#include "DYNModelVoltageMeasurementsUtilities.hpp"
+#include "DYNModelConstants.h"
+#include "DYNElement.h"
+#include "DYNMacrosMessage.h"
+#include "DYNElement.h"
+#include "DYNCommonModeler.h"
+#include "DYNTrace.h"
+#include "DYNVariableForModel.h"
+#include "DYNParameter.h"
+
+using std::vector;
+using std::string;
+using std::map;
+using std::stringstream;
+
+using boost::shared_ptr;
+
+using parameters::ParametersSet;
+
+/**
+ * @brief ModelVoltageMeasurementsUtilitiesFactory getter
+ *
+ * @return A pointer to a new instance of ModelVoltageMeasurementsUtilitiesFactory
+ */
+extern "C" DYN::SubModelFactory* getFactory() {
+  return (new DYN::ModelVoltageMeasurementsUtilitiesFactory());
+}
+
+/**
+ * @brief ModelVoltageMeasurementsUtilitiesFactory destroy method
+ */
+extern "C" void deleteFactory(DYN::SubModelFactory* factory) {
+  delete factory;
+}
+
+/**
+ * @brief ModelVoltageMeasurementsUtilities getter
+ *
+ * @return A pointer to a new instance of ModelVoltageMeasurementsUtilities
+ */
+extern "C" DYN::SubModel* DYN::ModelVoltageMeasurementsUtilitiesFactory::create() const {
+  DYN::SubModel* model(new DYN::ModelVoltageMeasurementsUtilities());
+  return model;
+}
+
+/**
+ * @brief ModelVoltageMeasurementsUtilitiesFactory destroy method
+ */
+extern "C" void DYN::ModelVoltageMeasurementsUtilitiesFactory::destroy(DYN::SubModel* model) const {
+  delete model;
+}
+
+namespace DYN {
+
+/**
+ * @brief ModelVoltageMeasurementsUtilities model default constructor
+ *
+ */
+ModelVoltageMeasurementsUtilities::ModelVoltageMeasurementsUtilities() :
+ModelCPP("voltageMeasurementsUtilities"),
+nbConnectedInputs_(0) {
+}
+
+/**
+ * @brief ModelVoltageMeasurementsUtilities model initialization
+ *
+ */
+void
+ModelVoltageMeasurementsUtilities::init(const double /*t0*/) {
+  // Nothing to initalize here.
+}
+
+void
+ModelVoltageMeasurementsUtilities::initializeFromData(const boost::shared_ptr<DataInterface>& /*data*/) {
+  // not needed
+}
+
+/**
+ * @brief ModelVoltageMeasurementsUtilities model's sizes getter
+ *
+ * Get the sizes of the vectors and matrices used by the solver to simulate
+ * ModelVoltageMeasurementsUtilities instance. Used by @p ModelMulti to generate right size matrices
+ * and vector for the solver.
+ */
+void
+ModelVoltageMeasurementsUtilities::getSize() {
+  sizeF_ = 0;  // No dynamics
+  sizeY_ = nbConnectedInputs_;  // All voltage inputs and their boolean activity values
+  sizeZ_ = nbConnectedInputs_;
+  sizeG_ = 0;
+  sizeMode_ = 1;
```

Comment:
I think sizeMode = 0 should be fine as we do not change mode
@rosiereflo do you confirm?